### PR TITLE
[Fix] [Bug] Fix bugs such as missing status and workflow when creating and modifying task definitions

### PR DESCRIPTION
**Pull request by jinyanhui2008**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by caishunfeng** [comment]
@jinyanhui2008 please add some description of this pr.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java [hunk 1]
```diff
@@ -506,28 +516,46 @@ public Map<String, Object> updateTaskWithUpstream(User loginUser, long projectCo
             ProcessTaskRelation taskRelation = upstreamTaskRelations.get(0);
             List<ProcessTaskRelation> processTaskRelations = processTaskRelationMapper.queryByProcessCode(projectCode, taskRelation.getProcessDefinitionCode());
             List<ProcessTaskRelation> processTaskRelationList = Lists.newArrayList(processTaskRelations);
-            List<ProcessTaskRelation> relationList = Lists.newArrayList();
+//            List<ProcessTaskRelation> relationList = Lists.newArrayList();
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java]:
```suggestion

```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java [hunk 2]
```diff
@@ -506,28 +516,46 @@ public Map<String, Object> updateTaskWithUpstream(User loginUser, long projectCo
...
             for (ProcessTaskRelation processTaskRelation : processTaskRelationList) {
                 if (processTaskRelation.getPostTaskCode() == taskCode) {
                     if (queryUpStreamTaskCodeMap.containsKey(processTaskRelation.getPreTaskCode()) && processTaskRelation.getPreTaskCode() != 0L) {
                         queryUpStreamTaskCodeMap.remove(processTaskRelation.getPreTaskCode());
                     } else {
                         processTaskRelation.setPreTaskCode(0L);
                         processTaskRelation.setPreTaskVersion(0);
-                        relationList.add(processTaskRelation);
+//                        relationList.add(processTaskRelation);
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java]:
```suggestion

```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java [hunk 3]
```diff
@@ -506,28 +516,46 @@ public Map<String, Object> updateTaskWithUpstream(User loginUser, long projectCo
...
                     }
                 }
             }
-            processTaskRelationList.removeAll(relationList);
+//            processTaskRelationList.removeAll(relationList);
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java]:
```suggestion

```

File: dolphinscheduler-ui/src/views/projects/task/definition/use-task.ts [hunk 4]
```diff
@@ -88,7 +88,9 @@ export function useTask(projectCode: number) {
 
   const onEditTask = async (row: IRecord, readonly: boolean) => {
     const result = await queryTaskDefinitionByCode(row.taskCode, projectCode)
-    task.taskData = { ...result, processName: row.processDefinitionCode }
+    // debugger
+    task.taskData = {...result, processName: row.processDefinitionCode, upstreamTaskMapKey: row.upstreamTaskMapKey}
+    // task.taskData = { ...result, processName: row.processDefinitionCode }
```
Comment (ruanwenjun) [inline dolphinscheduler-ui/src/views/projects/task/definition/use-task.ts:93]:
```suggestion

```

**Review by ruanwenjun — COMMENTED** [review]
If you change the front-end code just for debug, please remove these changes.

**Comment by jinyanhui2008** [comment]
Repair task definition function:

1. Modification of task definition may lead to loss of workflow definition and status

2. Missing predecessors when modifying task definitions

3. Creating and modifying task definitions will lose workflow definitions of other task definitions

**Pull request closed without merging**